src/http/one/TeChunkedParser.cc

```diff
@@ -158,8 +166,10 @@ Http::One::ChunkExtensionValueParser::Ignore(Tokenizer &tok, const SBuf &extName
 /// Parses a single chunk-ext list element:
 /// chunk-ext = *( BWS ";" BWS chunk-ext-name [ BWS "=" BWS chunk-ext-val ] )
 void
-Http::One::TeChunkedParser::parseOneChunkExtension(Tokenizer &tok)
+Http::One::TeChunkedParser::parseOneChunkExtension(Tokenizer &callerTok)
 {
+    auto tok = callerTok;
+
     ParseBws(tok); // Bug 4492: ICAP servers send SP before chunk-ext-name
 
     const auto extName = tok.prefix("chunk-ext-name", CharacterSet::TCHAR);
```

Comment:
We must also update callerTok for valueless chunk-ext (see an early `return` a few lines below). According to the chunk-ext ABNF, BWS after chunk-ext-name may not belong to chunk-ext[^1], so we have to update callerTok here, after successfully parsing chunk-ext-name (and before unconditionally consuming BWS). Here is one way to do that:

[^1]: BWS after chunk-ext-name does not belong to chunk-ext if this chunk-ext has no optional `BWS "=" BWS chunk-ext-val` component.

```suggestion
    const auto extName = tok.prefix("chunk-ext-name", CharacterSet::TCHAR);
    callerTok = tok; // in case we determine that this is a valueless chunk-ext
```